PF/Assignment1/RoomAllotmentMain.java

```diff
@@ -0,0 +1,95 @@
+import java.util.Scanner;
+
+//main class to allot rooms
+public class RoomAllotmentMain {
+
+	// main function
+	public static void main(String[] args) {
+		HashMap hm = new HashMap();
+		Scanner sc = new Scanner(System.in);
+		char ch;
+		do {
+			System.out.println("\nHash Table Operations\n");
+			System.out.println("1. book room ");
+			System.out.println("2. remove");
+			System.out.println("3. room status");
+			System.out.println("4. clear");
+			System.out.println("5. status of all rooms");
+
+			int choice;
+			do {// Validation
+
+				while (!sc.hasNextInt()) {
+					System.out
+							.println("That's not a number!Please enter again");
+					sc.next(); // this is important!
+				}
+				choice = sc.nextInt();
+			} while (choice <= 0);
+
+			switch (choice) {
+			case 1:
+
+				System.out.println("enter your name");
+
+				String name = sc.next();
+				System.out.println("Enter your age");
+				int age;
+				do {// Validation
+
+					while (!sc.hasNextInt()) {
+						System.out
+								.println("That's not a number!Please enter again");
+						sc.next(); // this is important!
+					}
+					age = sc.nextInt();
+				} while (age <= 0);
+				hm.put(age, name);
```

Comment:
ACTION: A HashMap contains values based on the key. Key should be unique in hashmap. so according to program-me age can be duplicate for two person.